Sources/_Runtime/Utilities/ImageInspection.swift

```diff
@@ -0,0 +1,126 @@
+//===----------------------------------------------------------*- swift -*-===//
+//
+// This source file is part of the Swift Argument Parser open source project
+//
+// Copyright (c) 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+@_implementationOnly import Foundation
+
+extension NSLock {
+  func withLock<T>(_ closure: () throws -> T) rethrows -> T {
+    lock()
+    defer { unlock() }
+    return try closure()
+  }
+}
+
+let conformanceLock = NSLock()
+var conformances:
+  [ContextDescriptor: [ContextDescriptor: [ConformanceDescriptor]]] = [:]
+
+@_cdecl("registerProtocolConformances")
+func registerProtocolConformances(_ section: UnsafeRawPointer, size: Int) {
+  // This section is a list of relative pointers.
+  for i in 0 ..< size / MemoryLayout<Int32>.size {
```

Comment:
Should (can?) we use `UnsafeRawBufferPointer` here instead of `UnsafeRawPointer` and `size`?

Reply:
We could, but since `UnsafeRawBufferPointer` isn't representable in C, we'd have to do construction of a mock buffer over on the C side, and on the Swift side we'd have to do some bit casting to eventually get the buffer pointer.

Reply:
The raw buffer would also be a weird interface because we're stepping by an `Int32` at a time, but don't want to bind the memory to `Int32`. I might approach this by just iterating over the pointers, e.g.:

```swift
for start in stride(from: section, to: section + size, by: MemoryLayout<Int32>.stride) { ... }
```